### PR TITLE
Improve population of potential relays and cleaning up potential relays

### PR DESCRIPTION
**Pull request by robertkiel**
Changes:
- event signature `addPublicNode` -> from `peer: Multiaddr` to `{ id: PeerId, multiaddrs: Multiaddr[] }`
- remove all addresses of offline nodes
- store potential relays grouped by their PeerId 

Breaking change:
- always require a PeerId when announcing a new public node

### CODE REVIEW

File: src/base/listener.spec.ts [hunk 1]
```diff
@@ -138,9 +145,11 @@ async function startNode(
     undefined
   )
 
-  await waitUntilListening(listener, new Multiaddr(`/ip4/127.0.0.1/tcp/0/p2p/${peerId.toB58String()}`))
+  process.nextTick(() =>
+    waitUntilListening(listener, new Multiaddr(`/ip4/127.0.0.1/tcp/0/p2p/${peerId!.toB58String()}`))
+  )
```
Comment (andreykiryushkin) [inline src/base/listener.spec.ts:150]: Can you explain what is the technical difference here?

File: src/base/listener.spec.ts [hunk 2]
```diff
@@ -205,20 +212,20 @@ describe('check listening to sockets', function () {
...
-    node.publicNodesEmitter.emit(
-      `addPublicNode`,
-      new Multiaddr(`/ip4/127.0.0.1/tcp/${relay.listener.getPort()}/p2p/${relay.peerId.toB58String()}`)
-    )
+    node.publicNodesEmitter.emit(`addPublicNode`, {
+      id: relay.peerId,
+      multiaddrs: [new Multiaddr(`/ip4/127.0.0.1/tcp/${relay.listener.getPort()}/p2p/${relay.peerId.toB58String()}`)]
+    })
 
     // Checks that relay and STUN got contacted, otherwise timeout
```
Comment (andreykiryushkin) [inline src/base/listener.spec.ts:230]: No timeout though

File: src/base/listener.spec.ts (hunk 1 above)
Comment (robertkiel) [reply src/base/listener.spec.ts:150]: Was following the Node.js documentation here ( https://nodejs.org/api/events.html#events_events_on_emitter_eventname_options ) and figured out that the previous version could lead to an unwanted control flow.

The new version (using process.nextTick) attaches the listener (using events.on) and the starts the program flow that generates the events (using process.nextTick) *asynchronously*

File: src/base/listener.spec.ts (hunk 2 above)
Comment (robertkiel) [reply src/base/listener.spec.ts:230]: thx, good catch

**Review by andreykiryushkin — APPROVED** [review]

**Pull request merged**